tardis/apps/publication_workflow/static/publication-actions/publication_actions.js

```diff
@@ -0,0 +1,269 @@
+     *
+     * The publicationId (a.k.a. experimentId) is passed to the view URL
+     * so the view method can set is_owner in the context passed to the
+     * tokens.html Django template.  The publicationId is also passed via the
+     * ngDialog's data to the PublicationTokensController instance.
+     */
+    vm.sharePublication = function () {
+        ngDialog.open({
+            template: '/apps/publication-workflow/tokens/' + vm.experimentId + '/',
+            data: {
+                'publicationId': parseInt(vm.experimentId)
+            },
+            closeByDocument: false,
+        });
+    };
+
+    /*
+     * Delete publication draft
+     */
+    vm.deletePublicationDraft = function () {
+        var confirmation = $mdDialog.confirm()
+            .title('Are you sure you want to delete Publication ID ' + vm.experimentId + '?')
+            .textContent('You cannot undo this action!')
+            .ok('Yes, delete it!')
+            .cancel('No, keep it.');
+  
+        $mdDialog.show(confirmation).then(function() {
+            $log.info('OK, deleting publication...');
+            $http.post('/apps/publication-workflow/publication/delete/' + vm.experimentId + '/', {})
+               .then(function (_response) {
+                   $log.debug("Publication deleted successfully.");
+                   $window.location.reload();
+               },
+               function(_response) {
+                   $log.debug("Failed to delete publication.");
+               });
+        }, function() {
+            $log.info('OK, keeping publication');
+        });
+    };
+    /*
+     * Mint DOI
+     */
+    vm.mintDOI = function () {
+        var confirmation = $mdDialog.confirm()
+            .title('Are you sure you want to mint a DOI for Publication ID ' + vm.experimentId + '?')
+            .textContent('You cannot undo this action!')
+            .ok('Yes, mint a DOI!')
+            .cancel('No, don\'t mint a DOI.');
+  
+        $mdDialog.show(confirmation).then(function() {
+            $log.info('OK, minting a DOI...');
+            vm.updating = true;
+            $http.post('/apps/publication-workflow/publication/mint_doi/' + vm.experimentId + '/', {})
+               .then(function (_response) {
+                   vm.updating = false;
+                   $log.debug("DOI minted successfully.");
+                   $window.location.reload();
+               },
+               function(_response) {
+                   vm.updating = false;
+                   $log.debug("Failed to mint DOI.");
+               });
+        }, function() {
+            vm.updating = false;
+            $log.info('OK, not minting a DOI');
+        });
+    };
+
+    /*
+     * Retract publication
+     */
+    vm.retractPublication = function () {
+        var confirmation = $mdDialog.confirm()
+            .title('Are you sure you want to retract Publication ID ' + vm.experimentId + '?')
```

Comment:
Codacy has a fix for the issue: [Strings must use doublequote.](https://app.codacy.com/gh/mytardis/mytardis/pullRequest?prid=7933951)
```suggestion
            .title("Are you sure you want to retract Publication ID " + vm.experimentId + '?')
```